### PR TITLE
JSON Summary output feature

### PR DESCRIPTION
**Pull request by matteocargnelutti**
Implements https://github.com/harvard-lil/scoop/issues/136.

- Added a `Scoop.summary()` method for generating a summary object. Can be invoked at any point during capture.
- Added a `--json-summary-output` CLI option for generating a summary at the end of the capture process and saving it to disk as JSON.

---

Profile of the summary object:
```javascript
/**
 * @typedef {Object} ScoopCaptureSummary
 * @property {int} state
 * @property {object} states - Possible values of Scoop.state
 * @property {string} targetUrl
 * @property {boolean} targetUrlIsWebPage
 * @property {ScoopOptions} options
 * @property {string} startedAt - ISO-formated date
 * @property {string[]} blockedRequests
 * @property {string[]} noArchiveUrls
 * @property {string[]} exchangeUrls
 * @property {?string} captureIp
 * @property {?string} userAgent
*/
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,8 +167,9 @@ More info: https://github.com/harvard-lil/scoop
 
 Options:
   -v, --version                                          Display Scoop and Scoop CLI version.
-  -o, --output <string>                                  Output path. (default: "./archive.wacz")
+  -o, --output <string>                                  Output path. (default: "[directory]/archive.wacz")
```
Comment (bensteinberg) [inline README.md:170]: I thought `.` was clearer than `[directory]` (unless we get to set `directory` somewhere).

File: README.md [hunk 2]
```diff
@@ -194,8 +194,7 @@ Options:
   --run-site-specific-behaviors <bool>                   Should Scoop run site-specific capture behaviors? (via: browsertrix-behaviors) (choices: "true", "false", default: "true")
   --headless <bool>                                      Should Chrome run in headless mode? (choices: "true", "false", default: "true")
   --user-agent-suffix <string>                           If provided, will be appended to Chrome's user agent. (default: "")
-  --blocklist <string>                                   If set, replaces Scoop's default list of url patterns and IP ranges Scoop should not capture. Coma-separated. Example:
-                                                         "/https?://localhost/,0.0.0.0/8,10.0.0.0".
+  --blocklist <string>                                   If set, replaces Scoop's default list of url patterns and IP ranges Scoop should not capture. Coma-separated. Example: "/https?://localhost/,0.0.0.0/8,10.0.0.0".
```
Comment (bensteinberg) [inline README.md:197]:
```suggestion
  --blocklist <string>                                   If set, replaces Scoop's default list of url patterns and IP ranges Scoop should not capture. Comma-separated. Example: "/https?://localhost/,0.0.0.0/8,10.0.0.0".
```

**Review by bensteinberg — APPROVED** [review]
🔥

File: README.md (hunk 1 above)
Comment (matteocargnelutti) [reply README.md]: That's fair!

**Comment by matteocargnelutti** [comment]
Thanks @bensteinberg !

**Pull request merged**